File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    let map = new Map();
    for (let i = 0; i < nums.length; i++) {
       if (map.has(nums[i])) {
           map.set(nums[i], true);
       } else {
           map.set(nums[i], false);
       }
    }

    for (let val of map.values()) {
        if (val) return true;
    }
    return false;
};

File: 0217-contains-duplicate/0217-contains-duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        mp = {}
        for x in nums:
            if x in mp:
                mp[x] += 1
            else:
                mp[x] = 1
        print(mp)
        for value in mp.values():
            if value > 1:
                return True
            
        return False